editor/components/post-featured-image/index.js

```diff
@@ -15,14 +15,20 @@ import { MediaUploadButton } from '@wordpress/blocks';
  * Internal dependencies
  */
 import './style.scss';
-import { getEditedPostAttribute } from '../../selectors';
+import { getCurrentPostType, getEditedPostAttribute } from '../../selectors';
 import { editPost } from '../../actions';
 
-function PostFeaturedImage( { featuredImageId, onUpdateImage, onRemoveImage, media } ) {
+//used when labels from post tyoe were not yet loaded or when they are not present.
+const DEFAULT_SET_FEATURE_IMAGE_LABEL = __( 'Set featured image' );
```

Comment:
Minor: We may not want to assume that `__` is a trivial operation, if we could avoid calling it as a side-effect of the module being imported (i.e. before we even know that we'll render it). Further, if we ever wanted to support changing locale in the current page session, these would always show the original value (vs. a forceful re-render of the whole app calling to get the latest value).

On second thought, if we _do_ consider `__` as non-trivial, we also have a similar problem that re-renders may occur frequently. Maybe we want to optimize it to be as trivial as possible, in which case it's not as big an issue to assign here.

Reply:
I opted for this version because I saw other similar usages e.g: TABLE_CONTROLS in table-block.js, or ALIGNMENT_CONTROLS in alignment-toolbar. But your point: "Further, if we ever wanted to support changing locale in the current page session", is really important not necessarily in this case (because we will get the labels from the API), but in other cases, it may be problematic. I think in the interest of allowing local changes without full refresh we may move __ calls into the render. But even that would be problematic, because there is no guarantee the refresh would happen (unless another prop changes).